libpromises/bootstrap.c

```diff
@@ -177,10 +198,57 @@ char *ReadPolicyServerFile(const char *workdir)
     }
     else
     {
+        Log(LOG_LEVEL_VERBOSE, "Could not fopen(%s).", filename);
         return NULL;
     }
 }
 
+bool ParsePolicyServerFile(const char *workdir, char** host, char** port)
+{
+    // *host and *port are output,
+    // these are allocated using xstrdup and need to be freed!
+    char* contents = ReadPolicyServerFile(workdir);
+    if (contents == NULL)
+    {
+        return false;
+    }
+    (*host) = NULL;
+    (*port) = NULL;
+
+    ParseHostPort(contents, host, port);
+    (*host) = xstrdup(*host);
+    if (*port != NULL)
+    {
+        (*port) = xstrdup(*port);
+    }
+    free(contents);
+    return true;
+}
+
+/* Combines reading the policy server file, parsing and lookup (host->ip)
+ * Output data is stored in ipaddr and port
+ * @return: false if either file read or ip lookup failed.
+ * */
+bool LookUpPolicyServerFile(const char *workdir, char** ipaddr, char** port)
+{
+    char* host;
+    bool file_read = ParsePolicyServerFile(workdir, &host, port);
+    if (file_read == false)
+    {
+        return false;
+    }
+    char tmp_ipaddr[CF_MAX_IP_LEN];
+    if (Hostname2IPString(tmp_ipaddr, host, sizeof(tmp_ipaddr)) == -1)
+    {
+        Log(LOG_LEVEL_ERR,
+            "LookUpPolicyServerFile: ERROR, could not resolve '%s'", host);
```

Comment:
LOG_LEVEL_ERR is very user visible, try to lose the `LookUpPolicyServerFile` hint. Also the "ERROR" string is automatically printed, so it is redundant in the message. I would just print `Unable to resolve policy server host: %s`.